src/interpreter/initialization.rs

```diff
@@ -88,20 +88,25 @@ impl<S> Interpreter<S> {
 
     // compute the initial free balances for each asset type
     pub(crate) fn initial_free_balances(&self) -> Result<HashMap<AssetId, Word>, InterpreterError> {
+        let base_asset = AssetId::default();
```

Comment:
Orthogonal to this PR, but I'm thinking the base asset's ID should be a parameter in the VM. It's not necessarily guaranteed to be `AssetId::default()` always. Both the specs and the implementation should be modular in this regard. Can you file an issue to track changing?